src/Foundatio.Redis/Cache/RedisCacheClient.cs

```diff
@@ -88,14 +87,43 @@ await foreach (var key in server.KeysAsync().ConfigureAwait(false))
         public async Task<int> RemoveByPrefixAsync(string prefix) {
             await LoadScriptsAsync().AnyContext();
 
+            const int chunkSize = 2500;
+            string regex = prefix + "*";
             try {
-                var result = await Database.ScriptEvaluateAsync(_removeByPrefix, new { keys = prefix + "*" }).AnyContext();
-                return (int)result;
+                int total = 0;
+                int index = 0;
+
+                (int cursor, string[] keys) = await ScanKeysAsync(regex, index, chunkSize).AnyContext();
+
+                while (keys.Length != 0 || index < chunkSize) {
+                    total += await RemoveAllAsync(keys).AnyContext();
+                    index += chunkSize;
+                    (cursor, keys) = await ScanKeysAsync(regex, cursor, chunkSize).AnyContext();
```

Comment:
Wouldn't this fail if there were more than 2500 batches?

Reply:
I'm not sure why it would fail if there were more than 2500 batches.

I've tested with 10.000.000 entries, it worked fine. 

(Except I had to split the parameters passed to SetAllAsync by batches of 1.000.000 as the current implementation does not support bigger enumerables - the ~1M limit is a Redis limit).

Reply:
I was thinking about the logic of `keys.Length != 0 || index < chunkSize` it didn't seem like the second part was correct.